### PR TITLE
NAS-132676 / 25.04 / Only redact secrets when the method return value is passed to the user

### PR DESCRIPTION
**Pull request by themylogin**
Previously, `call_with_audit` was redacting the secrets. This caused issues when its result was passed to the API versions conversion code (as it was trying to validate `port="********"` as integer).

### CODE REVIEW

**Comment by themylogin** [comment]
time 4:00

**Comment by themylogin** [comment]
time 2:00

File: tests/api2/test_legacy_websocket.py [hunk 1]
```diff
@@ -1,23 +1,70 @@
...
+            for i in range(1, 21):
+                try:
+                    c.call("auth.login_ex", {
+                        "mechanism": "PASSWORD_PLAIN",
+                        "username": t.username,
+                        "password": t.password,
+                    })
+                except ClientException as e:
+                    if e.errno == errno.EBUSY:
+                        time.sleep(61)
```
Comment (yocalebo) [inline tests/api2/test_legacy_websocket.py:57]: No, this is too much time wasted. You can make a call to `rate.limit.cache_clear` so we don't have to sleep here.

**Comment by themylogin** [comment]
http://jenkins.eng.ixsystems.net:8080/job/tests/job/api_tests/2075/

**Review by creatorcary — APPROVED** [review]

**Comment by themylogin** [comment]
@creatorcary this test was fixed after this branch was created

**Comment by themylogin** [comment]
@anodos325 ping

**Comment by themylogin** [comment]
@anodos325 ping

**Comment by anodos325** [comment]
```
  {
    "id": 14,
    "method": "zfs.dataset.bulk_process",
    "arguments": [
      "check_key",
      "********"
    ],
    "transient": false,
    "description": null,
    "abortable": false,
    "logs_path": null,
    "logs_excerpt": null,
    "progress": {
      "percent": 100,
      "description": "",
      "extra": null
    },
    "result": null,
    "result_encoding_error": "TypeError(\"Middleware.dump_result() missing 1 required positional argument: 'result'\")",
    ```
    I installed in test VM and it appears `core.get_jobs` may be broken.

**Comment by anodos325** [comment]
Looks like core.get_jobs may be broken by these changes.

**Review by anodos325 — APPROVED** [review]

**Pull request merged**